source/PlayerInfo.cpp

```diff
@@ -4003,6 +4037,31 @@ void PlayerInfo::CreateMissions()
 
 
 
+// New random stock is generated each time you land on a planet.
+void PlayerInfo::CreateRandomStock()
+{
+	int day = GetDate().DaysSinceEpoch();
+	for(const auto &rStock : planet->OutfitRandomStock())
+		for(const auto &stockItem : *rStock)
+			if(!planet->Outfitter().Has(stockItem.item) && Random::Int(100) < stockItem.probability)
```

Comment:
Having a random item can make sense even if the item is normally present in the shop, if the random stock has some depreciation. This would allow having a limited amount of items for lower price, and unlimited amount of normally priced ones.